src/src/Globals/Plugins.cpp

```diff
@@ -243,24 +243,26 @@ void post_I2C_by_taskIndex(taskIndex_t taskIndex, deviceIndex_t DeviceIndex) {
 /**
  * Call the plugin of 1 task for 1 function, with standard EventStruct and optional command string
  */
-bool PluginCallForTask(taskIndex_t task, byte Function, EventStruct *TempEvent, String& command, bool unConditional = false) {
+bool PluginCallForTask(taskIndex_t taskIndex, byte Function, EventStruct *TempEvent, String& command, EventStruct *event = nullptr) {
   bool retval = false;
-  if (unConditional || (Settings.TaskDeviceEnabled[task] && validPluginID_fullcheck(Settings.TaskDeviceNumber[task])))
+  if (Settings.TaskDeviceEnabled[taskIndex] && validPluginID_fullcheck(Settings.TaskDeviceNumber[taskIndex]))
   {
-    if (unConditional || Settings.TaskDeviceDataFeed[task] == 0) // these calls only to tasks with local feed
+    if (Settings.TaskDeviceDataFeed[taskIndex] == 0) // these calls only to tasks with local feed
     {
-      const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(task);
+      const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(taskIndex);
       if (validDeviceIndex(DeviceIndex)) {
-        TempEvent->TaskIndex    = task;
-        TempEvent->BaseVarIndex = task * VARS_PER_TASK;
+        TempEvent->setTaskIndex(taskIndex);
         TempEvent->sensorType   = Device[DeviceIndex].VType;
+        if (event != nullptr) {
+          TempEvent->OriginTaskIndex = event->TaskIndex;
```

Comment:
Why do you need the OriginTaskIndex?

Reply:
I tried to merge over the changes you made, don't know if it is really needed. If not I can remove it.

Reply:
Ah OK, if it is indeed a change I made, then leave it :)

I only saw it as your changes, did not notice it was a rebase merge.